### PR TITLE
Web/UI: Create modal for sharing events to social media

### PR DESCRIPTION
**Pull request by Limarulanda**
- Currently displays after a user creates/modifies an event they've created using the EditEvent component:

image

After user selects save event:
image

image

- Twitter (temporarily changed link to point to dev in order to show image):
image

- Email:
image

### CODE REVIEW

File: TrashMob/client-app/src/components/EventManagement/ShareToSocialsModal.tsx [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+  const handleCopyLink = () => {
+    navigator.clipboard.writeText(eventLink);
+    setCopied(true);
+    setTimeout(() => {
+      setCopied(false);
+    }, 2000)
+  }
+
+  const getShareMsgContent = (provider: string) => {
```
Comment (chunlishao) [inline TrashMob/client-app/src/components/EventManagement/ShareToSocialsModal.tsx:37]: The ShareUrl.tsx contains urls to share on twitter and facebook. Can you reuse that to keep the message consistent when share in different pages? 

Share button available on MyDashboard and event detail pages too. Possibly we can change the share button to this share dialog without event create information to keep consistence.  What do you think?

**Comment by joebeernink** [comment]
Also, please check the build for linter errors... I'd like to keep the warnings to 0, so we have warnings as errors.

File: TrashMob/client-app/src/components/EventManagement/ShareToSocialsModal.tsx [hunk 2]
```diff
@@ -0,0 +1,158 @@
+import { useState } from "react";
+import { Modal, Button, Tooltip, OverlayTrigger } from 'react-bootstrap';
+import { Clipboard, GeoAltFill, Clock } from "react-bootstrap-icons";
+import { FacebookShareButton, FacebookIcon, TwitterShareButton, TwitterIcon, LinkedinShareButton, LinkedinIcon, WhatsappShareButton, WhatsappIcon, EmailShareButton, EmailIcon } from 'react-share';
+import Card from 'react-bootstrap/Card';
+import * as ToolTips from '../../store/ToolTips';
+
+interface ModalProps {
+  createdEvent: any;
```
Comment (super-ming) [inline TrashMob/client-app/src/components/EventManagement/ShareToSocialsModal.tsx:9]:
```suggestion
import EventData from '../Models/EventData'

interface ModalProps {
  createdEvent: EventData;
}
```

File: TrashMob/client-app/src/components/EventManagement/ShareToSocialsModal.tsx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+  const getShareMsgContent = (provider: string) => {
+    return `Join my next ${provider === 'twitter' ? "@TrashMobEco" : "TrashMob.eco"} event on ${eventDate} at ${eventTime} in ${props.createdEvent.city}.\n` +
+      `Sign up using the link for more details. Help me clean up ${props.createdEvent.city}!`
+  }
+
+  const EventLink = () => {
+    return (
+      <Card className="pr-2">
+        <Card.Body style={{ padding: '0px', backgroundColor: '#f0f0f1' }}>
```
Comment (super-ming) [inline TrashMob/client-app/src/components/EventManagement/ShareToSocialsModal.tsx:50]: Let's avoid inline styling when possible. The Bootstrap class for zero padding is `p-0`.

File: TrashMob/client-app/src/components/Pages/MyDashboard.tsx [hunk 4]
```diff
@@ -202,6 +206,19 @@ const MyDashboard: FC<MyDashboardProps> = (props) => {
         }
     }, [reloadEvents, props.currentUser, props.currentUser.id, props.isUserLoaded]);
 
+    useEffect(() => {
+        if (state?.newEventCreated && isEventDataLoaded) {
+            var myFilteredList = myEventList.filter(event => event.createdByUserId === props.currentUser.id)
```
Comment (super-ming) [inline TrashMob/client-app/src/components/Pages/MyDashboard.tsx:211]:
```suggestion 
 const myFilteredList = myEventList.filter(event => event.createdByUserId === props.currentUser.id)
```

**Pull request closed without merging**